src/Categories/Double/Core.agda

```diff
@@ -0,0 +1,297 @@
+      Sq≈ (sq ∘₂ᵥ (id₂ᵥ hT)) sq
+
+    identity₂ₕ² : ∀ {V₁ V₂} {v : V₁ ⇒ᵥ V₂} → Sq≈ (id₂ₕ v ∘₂ₕ id₂ₕ v) (id₂ₕ v)
+    identity₂ᵥ² : ∀ {H₁ H₂} {h : H₁ ⇒ₕ H₂} → Sq≈ (id₂ᵥ h ∘₂ᵥ id₂ᵥ h) (id₂ᵥ h)
+
+{-
+horizontal 2-associativity:
+
+        T₁  →  T₂  →  T₃  →  T₄
+        ↓   s₁  ↓   s₂  ↓   s₃  ↓
+        B₁  →  B₂  →  B₃  →  B₄
+-}
+    assocₕ :
+      {T₁ T₂ T₃ T₄ B₁ B₂ B₃ B₄ : Obj} →
+      {hT₁ : T₁ ⇒ₕ T₂} {hT₂ : T₂ ⇒ₕ T₃} {hT₃ : T₃ ⇒ₕ T₄} →
+      {hB₁ : B₁ ⇒ₕ B₂} {hB₂ : B₂ ⇒ₕ B₃} {hB₃ : B₃ ⇒ₕ B₄} →
+      {v₁  : T₁ ⇒ᵥ B₁} {v₂  : T₂ ⇒ᵥ B₂} {v₃  : T₃ ⇒ᵥ B₃} {v₄ : T₄ ⇒ᵥ B₄} →
+      (s₃ : Sq₂ hT₃ hB₃ v₃ v₄) →
+      (s₂ : Sq₂ hT₂ hB₂ v₂ v₃) →
+      (s₁ : Sq₂ hT₁ hB₁ v₁ v₂) →
+      Sq≈ ((s₃ ∘₂ₕ s₂) ∘₂ₕ s₁) (s₃ ∘₂ₕ (s₂ ∘₂ₕ s₁))
+
+    sym-assocₕ :
+      {T₁ T₂ T₃ T₄ B₁ B₂ B₃ B₄ : Obj} →
+      {hT₁ : T₁ ⇒ₕ T₂} {hT₂ : T₂ ⇒ₕ T₃} {hT₃ : T₃ ⇒ₕ T₄} →
+      {hB₁ : B₁ ⇒ₕ B₂} {hB₂ : B₂ ⇒ₕ B₃} {hB₃ : B₃ ⇒ₕ B₄} →
+      {v₁  : T₁ ⇒ᵥ B₁} {v₂  : T₂ ⇒ᵥ B₂} {v₃  : T₃ ⇒ᵥ B₃} {v₄ : T₄ ⇒ᵥ B₄} →
+      (s₃ : Sq₂ hT₃ hB₃ v₃ v₄) →
+      (s₂ : Sq₂ hT₂ hB₂ v₂ v₃) →
+      (s₁ : Sq₂ hT₁ hB₁ v₁ v₂) →
+      Sq≈ (s₃ ∘₂ₕ (s₂ ∘₂ₕ s₁)) ((s₃ ∘₂ₕ s₂) ∘₂ₕ s₁)
+
+{-
+vertical 2-associativity:
+
+        T₁  →  T₂
+        ↓   s₁  ↓
+        T₃  →  T₄
+        ↓   s₂  ↓
+        B₁  →  B₂
+        ↓   s₃  ↓
+        B₃  →  B₄
+
+-}
+    assocᵥ :
+      {T₁ T₂ T₃ T₄ B₁ B₂ B₃ B₄ : Obj} →
+      {hT₁ : T₁ ⇒ₕ T₂} {hT₂ : T₃ ⇒ₕ T₄}
+      {hB₁ : B₁ ⇒ₕ B₂} {hB₂ : B₃ ⇒ₕ B₄}
+      {vL₁  : T₁ ⇒ᵥ T₃} {vL₂  : T₃ ⇒ᵥ B₁} {vL₃  : B₁ ⇒ᵥ B₃} →
+      {vR₁  : T₂ ⇒ᵥ T₄} {vR₂  : T₄ ⇒ᵥ B₂} {vR₃  : B₂ ⇒ᵥ B₄} →
+      (s₃ : Sq₂ hB₁ hB₂ vL₃ vR₃) →
+      (s₂ : Sq₂ hT₂ hB₁ vL₂ vR₂) →
+      (s₁ : Sq₂ hT₁ hT₂ vL₁ vR₁) →
+      Sq≈ ((s₃ ∘₂ᵥ s₂) ∘₂ᵥ s₁) (s₃ ∘₂ᵥ (s₂ ∘₂ᵥ s₁))
+
+    sym-assocᵥ :
+      {T₁ T₂ T₃ T₄ B₁ B₂ B₃ B₄ : Obj} →
+      {hT₁ : T₁ ⇒ₕ T₂} {hT₂ : T₃ ⇒ₕ T₄}
+      {hB₁ : B₁ ⇒ₕ B₂} {hB₂ : B₃ ⇒ₕ B₄}
+      {vL₁  : T₁ ⇒ᵥ T₃} {vL₂  : T₃ ⇒ᵥ B₁} {vL₃  : B₁ ⇒ᵥ B₃} →
+      {vR₁  : T₂ ⇒ᵥ T₄} {vR₂  : T₄ ⇒ᵥ B₂} {vR₃  : B₂ ⇒ᵥ B₄} →
+      (s₃ : Sq₂ hB₁ hB₂ vL₃ vR₃) →
+      (s₂ : Sq₂ hT₂ hB₁ vL₂ vR₂) →
+      (s₁ : Sq₂ hT₁ hT₂ vL₁ vR₁) →
+      Sq≈ (s₃ ∘₂ᵥ (s₂ ∘₂ᵥ s₁)) ((s₃ ∘₂ᵥ s₂) ∘₂ᵥ s₁)
+
+{-
+interchange law:
+
+        T₁  →  T₂  →  T₃
+        ↓   s₁  ↓   s₂  ↓
+        M₁  →  M₂  →  M₃
+        ↓   s₃  ↓   s₄  ↓
+        B₁  →  B₂  →  B₃
+-}
+    interchange :
+      {T₁ T₂ T₃ M₁ M₂ M₃ B₁ B₂ B₃ : Obj} →
+      {hT₁ : T₁ ⇒ₕ T₂} {hT₂ : T₂ ⇒ₕ T₃} →
+      {hM₁ : M₁ ⇒ₕ M₂} {hM₂ : M₂ ⇒ₕ M₃} →
+      {hB₁ : B₁ ⇒ₕ B₂} {hB₂ : B₂ ⇒ₕ B₃} →
+      {vL₁  : T₁ ⇒ᵥ M₁} {vL₂  : M₁ ⇒ᵥ B₁} →
+      {vM₁  : T₂ ⇒ᵥ M₂} {vM₂  : M₂ ⇒ᵥ B₂} →
+      {vR₁  : T₃ ⇒ᵥ M₃} {vR₂  : M₃ ⇒ᵥ B₃} →
+      (s₄ : Sq₂ hM₂ hB₂ vM₂ vR₂) →
+      (s₃ : Sq₂ hM₁ hB₁ vL₂ vM₂) →
+      (s₂ : Sq₂ hT₂ hM₂ vM₁ vR₁)
+      (s₁ : Sq₂ hT₁ hM₁ vL₁ vM₁) →
+      Sq≈ ((s₄ ∘₂ₕ s₃) ∘₂ᵥ (s₂ ∘₂ₕ s₁)) ((s₄ ∘₂ᵥ s₂) ∘₂ₕ (s₃ ∘₂ᵥ s₁))
+  -- When a category is quantified, it is convenient to refer to the levels from a module,
```

Comment:
Turns out these things (that I know you took from Category's own definition) are barely used - best to leave them out.